move/axelar_gateway/sources/types/proof.move

```diff
@@ -66,19 +84,23 @@ public(package) fun recover_pub_key(
 /// Otherwise, the error `ELowSignaturesWeight` is raised.
 public(package) fun validate(
     self: &Proof,
-    message: vector<u8>
+    message: MessageToSign,
 ) {
     let signers = self.signers();
     let signatures = self.signatures();
     assert!(signatures.length() != 0, ELowSignaturesWeight);
 
     let threshold = signers.threshold();
     let mut total_weight: u128 = 0;
+    let mut signer_index = 0;
+    let message_bytes = bcs::to_bytes(&message);
 
     signatures.do_ref!(|signature| {
-        let pub_key = signature.recover_pub_key(&message);
+        let pub_key = signature.recover_pub_key(&message_bytes);
+
+        let (weight, index) = signers.find_signer_weight(signer_index, &pub_key);
```

Comment:
`find_signer_weight` -> `find_weight_by_pubkey_from` maybe. I'd also define it in this module. It's too specific to the logic here to exist in the other module